src/vast_client.js

```diff
@@ -0,0 +1,130 @@
+import { Storage } from './util/storage';
+import { Util } from './util/util';
+import { VASTParser } from './parser/parser';
+
+/**
+ * This class provides methods to fetch and parse a VAST document using VASTParser.
+ * In addition it provides options to skip consecutive calls based on constraints.
+ * @export
+ * @class VASTClient
+ */
+export class VASTClient {
+  /**
+   * Creates an instance of VASTClient.
+   * @param  {Number} cappingFreeLunch - The number of first calls to skip.
+   * @param  {Number} cappingMinimumTimeInterval - The minimum time interval between two consecutive calls.
+   * @param  {Storage} customStorage - A custom storage to use instead of the default one.
+   * @constructor
+   */
+  constructor(cappingFreeLunch, cappingMinimumTimeInterval, customStorage) {
+    this.cappingFreeLunch = cappingFreeLunch || 0;
+    this.cappingMinimumTimeInterval = cappingMinimumTimeInterval || 0;
+    this.defaultOptions = {
+      withCredentials: false,
+      timeout: 0
+    };
+    this.vastParser = new VASTParser();
+    this.util = new Util();
+    this.storage = customStorage || new Storage();
+
+    // Init values if not already set
```

Comment:
How can it be already set ? we are in the constructor

Reply:
Yes we are, but these three properties are retrieved from the `storage` through getter methods. Therefore it can be that we already have values for them at this point of execution.